### PR TITLE
EXP-28048: adding api sec documentation as first iteration for Financial Integration service

### PR DESCRIPTION
**Pull request by srikanthbabu133**
adding api sec documentation as first iteration for Financial Integration service

@sarraloew

### CODE REVIEW

**Comment by sarraloew** [comment]
Hi Srikanth, I ran this JSON through the swagger editor at https://editor.swagger.io/ and it has 6 errors, please review and fix those.

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 1]
```diff
@@ -0,0 +1,875 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "Financial Integration APIs help Concur clients move extract data from Concur into their systems of record.",
+    "version": "4",
+    "title": "Financial Integration API",
+    "termsOfService": "http://terms-of-service.url",
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: remove the termsOfService, contact, and license objects, we don't use them.

**Comment by sarraloew** [comment]
Sent a message to @srikanthbabu133 about request parameters not matching the markdown. The markdown is currently under review so if need be we can wait until the markdown is reviewed before doing the swagger cross-check.

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 2]
```diff
@@ -0,0 +1,867 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "Financial Integration APIs help Concur clients move extract data from Concur into their systems of record.",
+    "version": "4",
+    "title": "Financial Integration API"
+  },
+  "host": "us.api.concursolutions.com",
+  "basePath": "/financialintegration",
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: We're not displaying the basePath value anywhere, so please empty this out and include it in the URL for each resource.

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 3]
```diff
@@ -0,0 +1,867 @@
...
+        "systemId": {
+          "type": "string",
+          "description": "The ERP Id that acknowledged the document."
+        }
+      }
+    },
+    "Collection.Resource.FIDocument": {
+      "type": "object"
+    },
+    "PublishedPurgeMessage": {
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: This model doesn't appear anywhere in the markdown- is it used?

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 4]
```diff
@@ -0,0 +1,867 @@
...
+          "items": {
+            "$ref": "#/definitions/Link"
+          }
+        }
+      }
+    },
+    "Collection.FIDocument": {
+      "type": "object"
+    },
+    "PublishedMessage": {
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: This model doesn't appear anywhere in the markdown- is it used?

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 5]
```diff
@@ -0,0 +1,867 @@
...
+        },
+        "timeStamp": {
+          "type": "string"
+        },
+        "topic": {
+          "type": "string"
+        }
+      }
+    },
+    "Facts": {
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: This model doesn't appear anywhere in the markdown- is it used?

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 6]
```diff
@@ -0,0 +1,867 @@
...
+        },
+        "timeStamp": {
+          "type": "string"
+        },
+        "topic": {
+          "type": "string"
+        }
+      }
+    },
+    "DocumentRefreshQueueResponseItem": {
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: This model doesn't appear anywhere in the markdown- is it used?

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 7]
```diff
@@ -0,0 +1,867 @@
...
+            "type": "string"
+          }
+        },
+        "systemId": {
+          "type": "string",
+          "description": "The ERP Id that acknowledged the document."
+        }
+      }
+    },
+    "Collection.Resource.FIDocument": {
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json:365]: This model is empty- shouldn't it link to Resource.FIDocument?

**Review by sarraloew — CHANGES_REQUESTED** [review]

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 8]
```diff
@@ -0,0 +1,867 @@
...
+            ]
+          },
+          {
+            "name": "Authorization",
+            "in": "header",
+            "description": "Bearer JWTFromOAuthServiceHere",
+            "required": true,
+            "type": "string"
+          }
+        ],
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json:232]: The markdown includes a payload for this request, please add it here.

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 9]
```diff
@@ -0,0 +1,867 @@
...
+        "responses": {
+          "200": {
+            "description": "Ok",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/PostingConfirmationResponseItem"
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: Should this be the PaymentConfirmationResponseItem instead?

**Review by sarraloew — CHANGES_REQUESTED** [review]

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 10]
```diff
@@ -0,0 +1,867 @@
...
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/AcknowledgeResponseItem"
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: The markdown lists an item called AcknowledgeResponse, an array that contains multiple AcknowledgeResponseItem objects. The swagger only lists AcknowledgeResponseItem, which is correct?

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 11]
```diff
@@ -0,0 +1,970 @@
...
+            "WAS_RECALLED"
+          ]
+        },
+        "systemId": {
+          "type": "string",
+          "description": "The ERP Id that acknowledged the document."
+        }
+      }
+    },
+    "NotificationReceivedResponseItem": {
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: This model doesn't appear in the markdown, is it in use?

File: src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json [hunk 12]
```diff
@@ -0,0 +1,970 @@
...
+        }
+      }
+    },
+    "Collection.Resource.FIDocument": {
+      "type": "array",
+      "items": {
+          "$ref": "#/definitions/Resource.FIDocument"
+      }
+    },
+    "FiDocumentResponse": {
```
Comment (sarraloew) [inline src/api-explorer/v4-0/FinancialIntegrationService.swagger2.json]: This model isn't used anywhere in the markdown, is it in use?

**Review by sarraloew — CHANGES_REQUESTED** [review]

**Comment by srikanthbabu133** [comment]
@sarraloew please review, addressed your latest comments and updated the file with all changes.

**Comment by sarraloew** [comment]
Sent question to Tuan and Srikanth about the messageLanguage supported values.

**Review by sarraloew — APPROVED** [review]

**Comment by sarraloew** [comment]
:shipit:

**Pull request merged**